extensions/interactions/GraphInput/directives/graph-input-rules.service.ts

```diff
@@ -161,14 +161,32 @@ export class GraphInputRulesService {
 
     // Check that for every vertex from the first graph there is a vertex in
     // the second graph with the same sum of weights of outgoing edges.
-    var degrees1 = adj1.map(function(value) {
-      return value.reduce(function(prev, cur) {
+    var degrees1 = adj1.map((value) => {
+      return value.reduce((prev, cur) => {
+        if (prev === null && cur === null) {
+          return 0;
+        }
+        if (prev === null) {
+          return cur;
+        }
+        if (cur === null) {
+          return prev;
+        }
```

Comment:
Why is this needed?

Reply:
`prev` and `cur` values are of type `number | null`. The typescript checks do not allow adding (+) objects which are `null`. This is because the `+` operator is supposed to be used with strings or numbers

Reply:
IMO it would be cleaner to just initialize these values with 0 if they are null.
```suggestion
        prev = (prev === null) ? 0 : prev;
        curr = (curr === null) ? 0 : curr;
```